drivers/overlay/overlay.go

```diff
@@ -1580,6 +1580,12 @@ func (d *Driver) get(id string, disableShifting bool, options graphdriver.MountO
 
 	composeFsLayers := []string{}
 	composeFsLayersDir := filepath.Join(dir, "composefs-layers")
+	if inAdditionalStore {
+		// Similarly to getMergedDir, we need to make sure we use a private directory for
+		// the composefs layers if in an additional store.  Otherwise, we could end up attempting
+		// a mount without having an exclusive lock on the additional store.
+		composeFsLayersDir = filepath.Join(d.runhome, id, "composefs-layers")
```

Comment:
There's a lot of hard to follow logic for me happening here. When I'm reviewing this PR I went to ask "ok what is `dir` here" - and it's the thing that gives us `inAdditionalStore`.

Then we have `mergedDir` which is computed using `inAdditionalStore` - and is seemingly written to solve *exactly* this issue, it says:

```
	// To prevent races with other processes mounting or unmounting the layer,
	// use a private directory under the main store rundir, not the "merged" 
```

So...would things be cleaner if we did something like:

```
var idDir string
if inAdditionalStore {
   idDir = path.Join(d.runhome, id)
} else {
   idDir = dir
}
```

There's probably a better name than `idDir`.

Reply:
Or maybe `getMergedDir()` changes to `getMountDir(..., "merged")` and `getMountDir(..., "composefs-layers")

Reply:
thanks for the feedback.

Pushed a new version to refactor the `if inAdditionalStore { idDir = path.Join(d.runhome, id)} else {   idDir = dir }` in a separate function